test/collections/VectorQueueTest.ooc

```diff
@@ -86,6 +86,42 @@ VectorQueueTest: class extends Fixture {
 			}
 			queue free()
 		})
+		this add("Queue indexing", func {
+			queue := VectorQueue<Int> new(10)
+			queue enqueue(0)
+			queue enqueue(1)
+			queue enqueue(2)
+			queue enqueue(3)
+			queue enqueue(4)
+			queue enqueue(5)
+			version(debugTests)
+				for (i in 0..queue count)
+					queue[i] toString() println()
+			expect(queue[0], is equal to(0))
+			expect(queue[1], is equal to(1))
+			expect(queue[2], is equal to(2))
+			expect(queue[3], is equal to(3))
+			expect(queue[4], is equal to(4))
+			expect(queue[5], is equal to(5))
+		})
+		this add("Queue negative indexing", func {
+			queue := VectorQueue<Int> new(10)
+			queue enqueue(0)
+			queue enqueue(1)
+			queue enqueue(2)
+			queue enqueue(3)
+			queue enqueue(4)
+			queue enqueue(5)
+			version(debugTests)
+				for (i in 1..queue count + 1)
+					queue[-i] toString() println()
+			expect(queue[-1], is equal to(5))
+			expect(queue[-2], is equal to(4))
+			expect(queue[-3], is equal to(3))
+			expect(queue[-4], is equal to(2))
+			expect(queue[-5], is equal to(1))
+			expect(queue[-6], is equal to(0))
+		})
```

Comment:
It would be nice to have a test that ensures the Queue will still work even if you go across the "seam" of the bounding buffer, both with positive and negative indexing. To do that, you will need to add a number of things to the queue that is larger than its capacity, but remove items along way so that the count never exceeds capacity, or the Queue will just expand to accommodate more items.